Use clang-format-11 in code-formatting checker

TimoWilken opened the pull request:
- install and use clang-format-11
- pass `--style file` to clang-format to explicitly use `.clang-format` file in project root
- minor improvements to formatting and small speed-up
- removed unused code
- added clang-format version to message on how to reproduce result

TimoWilken committed:
Use clang-format-11 in code-formatting checker

- install and use clang-format-11
- pass --style file to clang-format to explicitly use .clang-format file in
  project root
- minor improvements to formatting and small speed-up
- removed unused code
- added clang-format version to message on how to reproduce result

ktf commented on .github/workflows/code-formatting.yml:
```diff
@@ -15,68 +16,69 @@ jobs:
         COMMIT_FILES=$(git diff --name-only $BASE_COMMIT | grep -ivE 'LinkDef|Utilities/PCG/')
-        [ "X$COMMIT_FILES" = X ] && { echo "No files to check" ; exit 0; }
-        RESULT_OUTPUT="$(git-clang-format-8 --commit $BASE_COMMIT --diff --binary `which clang-format-8` $COMMIT_FILES)"
+        if [ -n "$COMMIT_FILES" ]; then
```
Do you really need to change the rest of the logic to bump clang-format 11?

TimoWilken replied:
OK, I've undone most of the formatting changes. Is this better?

ktf replied:
yes. Could you move the `11` in a variable?

ktf replied:
Alternatively, can't we simply drop the `-11`?

TimoWilken replied:
I've put the -11 in a variable. If we drop the -11, it'll use clang-format-10 in the Ubuntu containers provided by GitHub.

ktf replied:
Sorry for being pedantic... Actually I think:

```
update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-11 100
```

is better. This way we do not need a custom environment variable and we can simply have `clang-format`. Am I missing something?

TimoWilken replied:
OK, sure! I've also redirected git-clang-format and removed the `--binary ...` options as they should be superfluous now.

TimoWilken committed:
Make clang-format version configurable

Add a CLANG_FORMAT global variable to set the version.

TimoWilken committed:
Use clang-format directly

Add "alternatives" symlinks, which remove the need for passing "--binary"
options all the time.
